### PR TITLE
feat(select): enhanced select

### PR DESCRIPTION
**Pull request by moog16**
related #697

### CODE REVIEW

File: packages/menu/index.tsx [hunk 1]
```diff
@@ -194,8 +187,8 @@ export {
   ListDivider as MenuListDivider,
   ListGroup as MenuListGroup,
   ListGroupSubheader as MenuListGroupSubheader,
-  ListItemGraphic as MenuListGraphic,
-  ListItemMeta as MenuListMeta,
+  ListItemGraphic as MenuListItemGraphic,
```
Comment (moog16) [inline packages/menu/index.tsx]: change this in readme

File: packages/select/EnhancedSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      isInvalid,
+    } = this.props;
+
+    const {'aria-expanded': ariaExpanded, selectedValue, selectedItem} = this.state;
+
+    const selectedTextAttrs: {[key: string]: string} = {};
+    if (required) {
+      selectedTextAttrs['aria-required'] = required.toString();
+    }
+    if (ariaExpanded && ariaExpanded !== 'false') {
```
Comment (moog16) [inline packages/select/EnhancedSelect.tsx]: convert 'false' and 'true' to constants.

**Comment by moog16** [comment]
Running into https://github.com/Microsoft/TypeScript/issues/28762. Updating to typescript@3.3.3 seems to fix issue.

File: packages/select/NativeSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  get classes() {
+    return classnames('mdc-select__native-control', this.props.className);
+  }
+
+  attachRef = (node: HTMLSelectElement | null) => {
+    const {innerRef} = this.props;
+
+    // https://github.com/facebook/react/issues/13029#issuecomment-410002316
+    // @ts-ignore this is acceptable according to the comment above
```
Comment (williamernest) [inline packages/select/NativeSelect.tsx]: Can you replace `this is acceptable according to the comment above` with the reason why it's required ? Maybe `incorrectly typed as readonly - see github comment`

File: packages/select/README.md [hunk 4]
```diff
@@ -49,11 +50,42 @@ class MyApp extends React.Component {
...
+}
+```
+
 #### Shorthand options
 
-If the option elements do not require anything unique, you can omit passing `this.props.children` and set the `option` attribute.
+If the option elements do not require anything unique, you can omit passing `this.props.children` and set the `options` prop.
 
-This shorthand option is most useful when you're select options come in the form of JSON. Typically when building a select, the options will be backed by an array of objects or strings (either from an API endpoint or a JSON File). If the label/value key names, you can run a `map` function over your options.
+This options shorthand is most useful when you're select options come in the form of JSON. Typically when building a select, the options will be backed by an array of objects or strings (either from an API endpoint or a JSON File). If the label/value key names, you can run a `map` function over your options.
```
Comment (williamernest) [inline packages/select/README.md]: `you're` -> `your`
Typically when -> Typically, when

`If the label/value key names, you can run a map function over your options`...... what?

**Pull request merged**